### PR TITLE
MGMT-7219: Add internal API to transform a cluster from day1 to day2

### PR DESCRIPTION
**Pull request by nmagnezi**
# Assisted Pull Request

## List all the issues related to this PR

- [x] New Feature
- [ ] Bug fix
- [ ] Tests
- [ ] Documentation
- [ ] CI/CD 

## What environments does this code impact?

- [ ] Automation (CI, tools, etc)
- [x] Cloud
- [x] Operator Managed Deployments
- [ ] None

## How was this code tested?

- [ ] assisted-test-infra environment
- [ ] dev-scripts environment
- [ ] Reviewer's test appreciated
- [ ] Waiting for CI to do a full test run
- [ ] Manual (Elaborate on how it was tested)
- [x] No tests needed

## Assignees

/cc @filanov 
/cc @danielerez 
/cc @rollandf 

## Checklist

- [x] Title and description added to both, commit and PR.
- [x] Relevant issues have been associated (see [CONTRIBUTING] guide)
- [ ] Reviewers have been listed
- [ ] This change does not require a documentation update (docstring, `docs`, README, etc)
- [ ] Does this change include unit-tests (note that code changes require unit-tests)

## Reviewers Checklist

- [ ] Are the title and description (in both PR and commit) meaningful and clear?
- [ ] Is there a bug required (and linked) for this change?
- [ ] Should this PR be backported?

[Kubernetes community documentation]: https://github.com/kubernetes/community/blob/master/contributors/guide/pull-requests.md#commit-message-guidelines
[CONTRIBUTING]: https://github.com/openshift/assisted-service/blob/master/CONTRIBUTING.md

### CODE REVIEW

File: internal/bminventory/inventory.go [hunk 1]
```diff
@@ -1559,6 +1560,60 @@ func (b *bareMetalInventory) GetClusterDefaultConfig(_ context.Context, _ instal
...
+	if cluster, err = common.GetClusterFromDB(tx, clusterID, common.UseEagerLoading); err != nil {
+		log.WithError(err).Errorf("failed to find cluster %s", clusterID)
+		if errors.Is(err, gorm.ErrRecordNotFound) {
+			return nil, common.NewApiError(http.StatusNotFound, err)
+		}
+		return nil, common.NewApiError(http.StatusInternalServerError, err)
+	}
+
+	apiVipDnsname := fmt.Sprintf("api.%s.%s", cluster.Name, cluster.BaseDNSDomain)
+	dbReply := b.db.Model(&common.Cluster{}).Where("id = ?", cluster.ID.String()).
```
Comment (filanov) [inline internal/bminventory/inventory.go:1598]: i think that this logic belong to cluster api.

Comment (nmagnezi) [reply internal/bminventory/inventory.go]: Done

**Review by filanov — APPROVED** [review]

**Pull request merged**